docs/02_getting-started/03_smart-contract-development/04_data-persistence.md

```diff
@@ -30,144 +30,125 @@ Open the file in your favorite editor.
 
 ## Step 3: Write an Extended Standard Class and Include EOSIO
 
-If you followed the previous tutorial, you created a hello world contract and learned the basics. The code snippet uses a similiar structure with a class named `addressbook`:
+In the previous [2.1: Hello World Contract](./01_hello-world.md) tutorial, you created a hello world contract and learned the basics. The code snippet below uses a similar pattern and creates a class named `addressbook`:
 
 ```cpp
 #include <eosio/eosio.hpp>
 
 using namespace eosio;
 
-class [[eosio::contract("addressbook")]] addressbook : public eosio::contract {
+class [[eosio::contract]] addressbook : public eosio::contract {
   public:
 
   private:
 
 };
 ```
 
-## Step 4: Create The Data Structure for the Table
+## Step 4: Create The Data Structure For The Address Book
 
-Before a table can be configured and instantiated, we need to define a struct that represents the data structure of the address book. Since it is an address book, the table will contain people, so create a `struct` called "person"
+In this step you will create the following:
+1. the C++ `person` struct which defines the structure of the objects stored in `addressbook` underlying `kv table`
+2. the `kv_address_table` type which defines the `kv table` where the `person` objects are stored
 
-```cpp
-struct person {};
-```
+### Step 4.1: Define The Person Structure
 
-When defining the structure of a multi_index table, you use a unique value as the primary key.
+The `addressbook` smart contract keeps a record for each person into a `key-value table` also referred to as `kv table`.  A `kv table` serves as an on-chain storage location which is organized as a table of rows where each row stores objects with the same definition.
```

Comment:
I don't think this is clear. Perhaps ...

The `addressbook` smart contract stores person records in a `key-value table` or `kv table`.  The `kv table` is an on-chain data store, organized in rows, where each row stores an object. Each object must be of the same type. (--- is this true???)

Reply:
yes it is true.
I will make changes based on your feedback.